test/writer.rst

```diff
@@ -649,8 +649,8 @@ Smart quotes, ellipses, dashes
 
 ‘He said, “I want to go.”’ Were you alive in the 70’s?
 
-Here is some quoted ‘``code``’ and a “`quoted
-link <http://example.com/?foo=1&bar=2>`__”.
+Here is some quoted ‘``code``’ and a `quoted
```

Comment:
Is there really any reason to do this flattening for Quoted?
The original lines in this test are perfectly valid RST, are they not?

Reply:
good point, i didn't realise that quotes are not RST inlines. i modified the `combine` function as you suggested